src/python/pants/backend/shell/util_rules/shell_command.py

```diff
@@ -121,9 +154,9 @@ async def _prepare_process_request_from_target(
         output_files=output_files,
         output_directories=output_directories,
         fetch_env_vars=shell_command.get(ShellCommandExtraEnvVarsField).value or (),
-        append_only_caches=None,
-        supplied_env_var_values=FrozenDict(supplied_env_var_values),
-        immutable_input_digests=FrozenDict(immutable_input_digests),
+        append_only_caches=merged_extras.append_only_caches,
+        supplied_env_var_values=FrozenDict(extra_env),
+        immutable_input_digests=FrozenDict(merged_extras.immutable_input_digests),
```

Comment:
isn't `merged_extras.immutable_input_digests` immutable already (so no need to wrap in `FrozenDict`?)

Reply:
Yup! Copy-pasta deleted :)